### ISSUE TITLE
Missing subdependency in testcontainers dependency

### ISSUE THREAD

**Issue by hansdesmet** [spring-io/initializr#1123]
When I add the testcontainers dependency,
I still have to add this dependency **manually** to pom.xml:
```xml
<dependency>
  <groupId>org.testcontainers</groupId>
  <artifactId>junit-jupiter</artifactId>
  <scope>test</scope>
</dependency>
```
in order to use @TestContainers and @Container in my JUnit test.
Code fragment:
```java
@Testcontainers
class MyTest {
 @Container
    private static final MySQLContainer<?> mySQL
    =new MySQLContainer<>("mysql:latest");
```

**Comment by snicoll** [comment]
Given that `testcontainers` provides JUnit 4 support out-of-the-box and that JUnit Jupiter is an optional module, I am tempted to switch to that. The net effect is that we'd stop removing it when a more concrete match has been found. 

We recommend the user of JUnit Jupiter for new project so that looks like a consistent move. Flagging for team attention to see what the rest of the team thinks.

**Comment by wilkinsona** [comment]
Agreed. The JUnit Jupiter integration is really useful and I think it makes sense for users to always have it by default.
